component.mk

```diff
@@ -1,4 +1,128 @@
-ARDUINO_CORE_LIBS := $(patsubst $(COMPONENT_PATH)/%,%,$(sort $(dir $(wildcard $(COMPONENT_PATH)/libraries/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/*/*/*/))))
+ARDUINO_CORE_LIBS := $(patsubst $(COMPONENT_PATH)/%,%,$(sort $(dir $(wildcard $(COMPONENT_PATH)/libraries/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/*/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/*/*/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/*/*/*/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/*/*/*/*/*/*/))))
+
+SPECIFIC_LIBS :=
+
+ifneq "y" "$(CONFIG_ARDUINO_SELECTIVE_ArduinoOTA)"
+MODULE := ArduinoOTA
+SPECIFIC_LIBS += $(patsubst $(COMPONENT_PATH)/%,%,$(sort $(dir $(wildcard $(COMPONENT_PATH)/libraries/$(MODULE)/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/$(MODULE)/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/$(MODULE)/*/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/$(MODULE)/*/*/*/*/)) $(dir $(wildcard $(COMPONENT_PATH)/libraries/$(MODULE)/*/*/*/*/*/))))
```

Comment:
can this be made into function that accepts module name as argument and not repeat this long line so many times :) that will help

Reply:
I think we both know it can. And also it can be driven by the list of folders and then we will have no need to re-code this file and only need to list components in Kconfig. It also needs time.